lib/buffer.js

```diff
@@ -264,8 +264,8 @@ function fromObject(obj) {
   }
 
   if (obj) {
-    if (isArrayBuffer(obj.buffer) || 'length' in obj ||
-        isSharedArrayBuffer(obj)) {
+    if ('length' in obj || isArrayBuffer(obj.buffer) ||
```

Comment:
yes. it would cause this one edge case to throw:
```js
Buffer.from({ 0: 'a', 1: 'b', length: undefined })
```
since this isn't documented or tested i feel like it's a valid minor version change.